sw/device/silicon_creator/rom/rom.c

```diff
@@ -155,7 +154,7 @@ static rom_error_t rom_init(void) {
   SEC_MMIO_WRITE_INCREMENT(kFlashCtrlSecMmioInit);
 
   // Initialize in-memory copy of the ePMP register configuration.
-  rom_epmp_state_init(&epmp, lc_state);
+  rom_epmp_state_init(&in_memory_epmp, lc_state);
```

Comment:
Now that we a standalone `epmp_state_t` variable in `static_critical_epmp_state_t`, let's:
- [ ] Remove the `epmp_state_t *` parameter from all epmp functions,
- [ ] Add a weak `epmp_state_t` symbol in the epmp library which will be overridden by the strong one defined in `static_critical_epmp_state_t` but allows the epmp library to be used as a standalone lib, e.g. for tests.

Reply:
What it would be the advantage of making the epmp lib to use the global variable?

I would prefer to keep it as it is, because epmp implementation is context free lib, which makes it easier to be reused by other applications even thought this means making a copy of the lib.

Reply:
The advantage would be consistency. This wouldn't affect it's re-usability since `epmp_state` is supposed to be a singleton, i.e. we don't have multiple states that we track. For tests, we could just depend on the library. For production, depending on `static_critical_epmp` would be enough to put the struct in the `.static_critical` section.